Src/ReadAccelGyroMagnetism.c

```diff
@@ -27,22 +26,22 @@ static const int CMD_TIMEOUT = 150;
 #define ACCEL_X_LOW_REGISTER_ADDR 0x28
 #define MAGNETO_X_LOW_REGISTER_ADDR 0x28
 
+#define ACCEL_SENSITIVITY 0.732 // Unit is mg/LSB
+#define GYRO_SENSITIVITY 17.50 // Unit is mdps/LSB
+#define MAGENTO_SENSITIVITY 0.14 // Unit is mgauss/LSB
+
 // Full Commands
-static const uint8_t ACTIVATE_GYRO_ACCEL_CMD = G1_CTRL_REGISTER_ADDR | WRITE_CMD_MASK | ACCEL_GYRO_MASK;
-// 011 11 0 00 -> ODR 119, 2000 DPS
-static const uint8_t ACTIVATE_GYRO_ACCEL_DATA = 0x68 | WRITE_CMD_MASK | ACCEL_GYRO_MASK;
+static const uint8_t ACTIVATE_GYRO_ACCEL_CMD = G1_CTRL_REGISTER_ADDR | WRITE_CMD_MASK;
+// 011 01 0 00 -> ODR 119, 500 DPS
+static const uint8_t ACTIVATE_GYRO_ACCEL_DATA = 0x68;	// ODR 119 has cutoff of 38Hz
 
-static const uint8_t SET_ACCEL_SCALE_CMD = XL6_CTRL_REGISTER_ADDR | WRITE_CMD_MASK | ACCEL_GYRO_MASK;
+static const uint8_t SET_ACCEL_SCALE_CMD = XL6_CTRL_REGISTER_ADDR | WRITE_CMD_MASK;
 // 011 01 0 00 -> ODR 119, +/- 16G
 static const uint8_t SET_ACCEL_SCALE_DATA = 0x68;
 
-static const uint8_t SET_MAGNETO_ODR_CMD = M1_CTRL_REGISTER_ADDR | WRITE_CMD_MASK | MAGNETO_MASK;
-// 0 00 100 0 0 -> temp comp disabled, low performance mode, 10 ODR, fast ODR disabled, self-test disabled
-static const uint8_t SET_MAGNETO_ODR_DATA = 0x10;
-
-static const uint8_t ACTIVATE_MAGNETO_CMD = M3_CTRL_REGISTER_ADDR | WRITE_CMD_MASK | MAGNETO_MASK;
-// 1 0 0 00 1 00 -> I2C Disable, Low power mode disabled, SPI read write enable, Continuous-conversion mode
-static const uint8_t ACTIVATE_MAGNETO_DATA = 0x84;
+static const uint8_t ACTIVATE_MAGNETO_CMD = M3_CTRL_REGISTER_ADDR | WRITE_CMD_MASK;
+// 1 0 0 00 0 00 -> I2C Disable, Low power mode disabled, SPI write enable, Continuous-conversion mode
+static const uint8_t ACTIVATE_MAGNETO_DATA = 0x80;
```

Comment:
So keep this the old way doesn't give values again right?

Reply:
No the magnetometer values are never right.